File: my-kotlin/mykotlin/src/starter/_02_MyNullCheck.java
package starter;

public class _02_MyNullCheck {

    public static void main(String[] args) {

        String str = "A";
        String nullStr = null;
        System.out.println(str == null ? -1 : str.length());
        System.out.println(nullStr == null ? -1 : str.length());
    }

    public boolean startsWithA1(String s) {
        if (s == null) {
            throw new IllegalArgumentException("s must not be null");
        }
        return s.startsWith("A"); // NPE possible
    }

    public Boolean startsWithA2(String s) {
        if (s == null) {
            return null;
        }
        return s.startsWith("A");
    }

    public boolean startsWithA3(String s) {
        if (s == null) {
            return false;
        }
        return s.startsWith("A");
    }


}

File: my-kotlin/mykotlin/src/starter/_02_MyNullCheck.kt
package starter

fun main() {

    // 1. Safe Call
    val str: String? = "A"
    val nullStr: String? = null
//    str.length // safe call
    str?.length

    // 2. Elvis 연산자
    // str == null ? -1 : str.length()
    println(str?.length ?: -1)
    println(nullStr?.length ?: -1)
}

fun startsWithA1(str: String?): Boolean{
    if (str == null) {
        throw IllegalArgumentException("str must not be null")
    }
    return str.startsWith("A")
}

fun startsWithA2(str: String?): Boolean?{
    if (str == null)
        return null
    return str.startsWith("A")
}

fun startsWithA3(str: String?): Boolean?{
    if (str == null)
        return false
    return str.startsWith("A")
}